crates/sui-graphql-rpc/src/types/chain_identifier.rs

```diff
@@ -4,33 +4,41 @@
 use crate::{
     data::{Db, DbConnection, QueryExecutor},
     error::Error,
+    server::watermark_task::ChainIdentifierLock,
 };
 use async_graphql::*;
-use diesel::{ExpressionMethods, QueryDsl};
+use diesel::{ExpressionMethods, OptionalExtension, QueryDsl};
 use sui_indexer::schema::checkpoints;
 use sui_types::{
     digests::ChainIdentifier as NativeChainIdentifier, messages_checkpoint::CheckpointDigest,
 };
 
-pub(crate) struct ChainIdentifier;
+#[derive(Clone, Copy, Debug, Default)]
+pub(crate) struct ChainIdentifier(pub(crate) NativeChainIdentifier);
```

Comment:
As per our earlier discussion, we should consider making this an `Option<NativeChainIdentifier>`, so a query that comes in before the chain identifier has been set can detect that and respond accordingly.